clang/include/clang/Frontend/CompilerInvocation.h

```diff
@@ -242,6 +252,56 @@ class CompilerInvocation : public CompilerInvocationRefBase,
   void generateCC1CommandLine(llvm::SmallVectorImpl<const char *> &Args,
                               StringAllocator SA) const;
 
+  /// Parse command line options that map to \p CASOptions.
+  static bool ParseCASArgs(CASOptions &Opts, const llvm::opt::ArgList &Args,
+                           DiagnosticsEngine &Diags);
+
+  /// Generate command line options from CASOptions.
+  static void GenerateCASArgs(const CASOptions &Opts,
+                              SmallVectorImpl<const char *> &Args,
+                              CompilerInvocation::StringAllocator SA);
+  /// @}
+  /// @name Option Subgroups
+  /// @{
+
+  CASOptions &getCASOpts() { return CASOpts; }
+  const CASOptions &getCASOpts() const { return CASOpts; }
+
+  AnalyzerOptionsRef getAnalyzerOpts() const { return AnalyzerOpts; }
```

Comment:
Do you know where these extra accessors came from? We should already inherit all of them (except CASOpts) from `CompilerInvocationRefBase` or `CompilerInvocationValueBase`.